src/services/database/__tests__/delete-health-record.integration.test.js

```diff
@@ -0,0 +1,114 @@
+  const HealthRecord = ModelFactory.getByName(healthRecordModelName);
+  const Message = ModelFactory.getByName(messageModelName);
+  // =====================================================================
+
+  // ========================= SET UP / TEAR DOWN ========================
+  beforeEach(async () => {
+    await HealthRecord.truncate();
+    await Message.truncate();
+    await ModelFactory.sequelize.sync({ force: true });
+  });
+
+  afterAll(async () => {
+    await ModelFactory.sequelize.close();
+  });
+  // =====================================================================
+
+  describe('deleteHealthRecordAndMessages', () => {
+    it('should delete the health record and messages successfully, given a valid conversation ID', async () => {
+      // given
+      const [messageId, conversationId] = generateMultipleUUID(2, false);
+      const fragmentMessageIds = generateMultipleUUID(5, false);
+      const nhsNumber = generateRandomNhsNumber();
+
+      // when
+      const deleteObject = S3Service.mockImplementation(() => {
+        return {
+          deleteObject: jest.fn().mockResolvedValueOnce(undefined),
+        };
+      });
+
+      await createEhrExtract({
+        conversationId,
+        messageId,
+        nhsNumber,
+        fragmentMessageIds,
+      });
+
+      await deleteHealthRecordAndMessages(conversationId);
+
+      const foundMessages = await findAllMessagesByConversationId(conversationId);
+      const healthRecord = await findHealthRecordByConversationId(conversationId);
+
+      // then
+      expect(foundMessages).toEqual([]);
+      expect(healthRecord).toBeNull();
+      expect(deleteObject).toBeCalledTimes(1);
+    });
+
+    it('should throw an error, given an invalid conversation ID', async () => {
+      // given
+      const conversationId = generateRandomUUID(false);
+
+      // when
+      S3Service.mockImplementation(() => {
+        return {
+          deleteObject: jest.fn().mockResolvedValueOnce(undefined),
+        };
+      });
+
+      // then
+      await expect(async () => deleteHealthRecordAndMessages(conversationId)).rejects.toThrow(
+        NoMessageFoundError
+      );
+    });
+
+    it('should throw an error, given an error occurs within the S3 process', async () => {
+      // given
+      const [messageId, conversationId] = generateMultipleUUID(2, false);
+      const fragmentMessageIds = generateMultipleUUID(5, false);
+      const nhsNumber = generateRandomNhsNumber();
+
+      // when
+      S3Service.mockImplementation(() => {
+        return {
+          deleteObject: jest.fn().mockRejectedValueOnce(new NoS3ObjectsFoundError()),
+        };
+      });
+
+      await createEhrExtract({
+        conversationId,
+        messageId,
+        nhsNumber,
+        fragmentMessageIds,
+      });
+
+      // then
+      await expect(async () => deleteHealthRecordAndMessages(conversationId)).rejects.toThrow(
```

Comment:
```suggestion
      await expect(deleteHealthRecordAndMessages(conversationId)).rejects.toThrow(
```
Same as above, the test still work fine without the `async () => `, but it's OK if you prefer to have that